components/radial-chart/src/radial-chart.ts

```diff
@@ -0,0 +1,293 @@
+    return this._legendPosition;
+  }
+  set legendPosition(value: 'right' | 'bottom') {
+    this._legendPosition = value;
+  }
+  private _legendPosition: 'right' | 'bottom' = 'right';
+
+  /** @internal Series data, <dt-radial-chart-series> */
+  @ContentChildren(DtRadialChartSeries) _radialChartSeries: QueryList<
+    DtRadialChartSeries
+  >;
+
+  /** @internal Overlay template */
+  @ContentChild(DtRadialChartOverlay, { static: false, read: TemplateRef })
+  _overlay: TemplateRef<{ $implicit: DtRadialChartOverlayData }>;
+
+  /** @internal Render data used to draw the SVG. */
+  _renderData: DtRadialChartRenderData[] = [];
+
+  /** @internal Destroy subject to clear subscriptions on component destroy. */
+  private readonly _destroy = new Subject<void>();
+
+  /** @internal The chart's width based on the viewport width. */
+  _width = 0;
+
+  /** @internal The chart's radius. */
+  get _radius(): number {
+    // tslint:disable-next-line: no-magic-numbers
+    return this._width / 2;
+  }
+
+  /**
+   * @internal
+   * Flag if a background shape should be rendered.
+   * Depends on difference of the given and the calculated max value.
+   */
+  _hasBackground = false;
+
+  /**
+   * @internal
+   * The background path is either a full circle or a ring.
+   * Depends on the chart type.
+   */
+  _backgroundPath: string;
+
+  /**
+   * @internal
+   * Sum of all series values.
+   */
+  _totalSeriesValue: number;
+
+  /**
+   * @internal
+   * Get viewbox data based on the radius and width.
+   * Adjust the viewbox to center the circle.
+   */
+  get _viewBox(): string {
+    // tslint:disable-next-line: no-magic-numbers
+    return `${-this._width / 2} ${-this._width / 2} ${this._width} ${
+      this._width
+    }`;
+  }
+
+  /** @internal The chart's inner radius based on the chart type and defined fraction. */
+  get _innerRadius(): number {
+    return this.type === 'pie' ? 0 : this._radius * DONUT_INNER_CIRCLE_FRACTION;
+  }
+
+  constructor(
+    private _elementRef: ElementRef<HTMLElement>,
+    private _viewportResizer: DtViewportResizer,
+    private _changeDetectorRef: ChangeDetectorRef,
+    private _sanitizer: DomSanitizer,
+  ) {}
+
+  /** AfterContentInit hook */
+  ngAfterContentInit(): void {
+    /**
+     * Fires every time a value within one of the series changes
+     * or the series data itself gets an update (one series is added or removed).
+     */
+    this._radialChartSeries.changes
+      .pipe(
+        takeUntil(this._destroy),
+        switchMap(() =>
+          this._radialChartSeries.length
+            ? combineLatest(
+                this._radialChartSeries.map(series => series._stateChanges),
+              )
+            : of(null),
+        ),
+      )
+      .subscribe((): void => {
+        this._updateRenderData();
+      });
+
+    /**
+     * Fires every time the viewport size changes and applies the new width of the chart.
+     */
+    this._viewportResizer
```

Comment:
can you combine that with the radialChartSeries changes?

Reply:
I don't know. Can I? When the series array gets an update, I only have to update my render data. If the viewport changes, I have to set a new value for the `_width` and then update the render data. So I think this is, why I have two observables here.

If you know, how to do it better, you have to explain this to me, please :)